robot_namespace: ""
camera_topic: "/camera_wrist/depth_registered/z_filtered/points/"
reference_frame: {
    global_frame: "world",
    robot_frame: "base_link",
}
#-M_PI/4 - M_PI/16 for ur5e robot and robotiq 2f 85 gripper
end_effector: {
    gripper_offset: -0.02,
    approach_dist: 0.10,
    retreat_dist: 0.10,
    grasping: {
        grasp_rot_x: 0,
        grasp_rot_y: 0.39269908169 ,
        grasp_rot_z: -0.981747704,
        grasp_rot_w: 1,
    },
    command_topic: "/r2f85_gripper_controller/gripper_cmd",
    # max_close_val: 0.85,
    max_close_val: 0.75,
}
sim_objects: {
    position: {
        x_min: 0.5125,
        x_max: 0.6125,
        y_min: -0.2000,
        y_max: 0.0000,
        z: 1.1,
    },
    orientation: {
        roll_min: 0.0,
        roll_max: 0.0,
        pitch_min: 0.0,
        pitch_max: 0.0,
        yaw_min: 0.0,
        yaw_max: 0.0,
    },
}
grasp_plan: {
    timeout_len: 10.0,
}
move_group: {
    planning_group: "manipulator",
    MaxVelocityScalingFactor: 1.0,
    GoalPositionTolerance: 0.001,
    GoalOrientationTolerance: 0.002,
    setPlannerId: "RRTConnect",
    jump_threshold: 5.0,
    eef_step: 0.01,
}
collision_list:
    - id: table
      position: [0.0, 0.0, -0.1]
      # orientation will be in radians
      orientation: [0.0, 0.0, 0.0]
      size: [2.0, 2.0, 0.2]
    - id: rear_wall
      orientation: [0.0, 0.0, 0.0]
      position: [1.1, 0.0, 1.0]
      size: [0.2, 2.0, 2.0]
filters: {
    euclidean_cluster_extraction: {
        clusterTolerance: 0.008,
        minClusterSize: 10,
        maxClusterSize: 25000,
    },
    conditional_outlier_removal: {
        field: "z",
        min: 0.0,
        max: 0.8,
    },
    passthrough: {
        x: {
            min: -1.125,
            max: -0.225,
        },
        y: {
            min: -0.3,
            max: 1.5,
        },
        z: {
            min: -0.2,
            max: 1,
        },
    },
    plane_segmentation: {
        maxIterations: 50,
        distanceThreshold: 0.01,
    },
    radius_outlier_removal: {
        searchRadius: 0.8,
        minNeighborRadius: 2,
    },
    statistical_outlier_removal: {
        meanK: 50,
        stdDevMulThresh: 1.0,
    },
    voxel_grid: {
        leaf_size_x: 0.008,
        leaf_size_y: 0.008,
        leaf_size_z: 0.008,
    },
}
